bears/general/AnnotationBear.py

```diff
@@ -88,66 +95,130 @@ def find_annotation_ranges(self,
             A dictionary containing the various ways to define multi-line
             comments in a language.
         :return:
-            Two tuples first containing a tuple of strings, the second a tuple
-            of comments.
+            Four tuples containing dictionary for singleline strings,
+            multiline strings, singleline comments and multiline comments
+            respectively.
         """
         text = ''.join(file)
-        strings_range = []
-        comments_range = []
+        singleline_string_range = []
+        multiline_string_range = []
+        singleline_comment_range = []
+        multiline_comment_range = []
         position = 0
+
         while position <= len(text):
 
             def get_new_position():
-                _range, end_position = self.get_range_end_position(
-                    file,
-                    filename,
-                    text,
-                    multiline_string_delimiters,
-                    position,
-                    self.get_multiline)
-                if end_position and _range:
-                    strings_range.append(_range)
+                end_position, start_delim, end_delim = (
+                    self.get_range_end_position(
+                        file,
+                        filename,
+                        text,
+                        multiline_string_delimiters,
+                        position,
+                        self.get_multiline))
+                if end_position:
```

Comment:
@rhemon This is exactly what I already mentioned in https://github.com/coala/coala-bears/pull/1332/files#r113981233

cc @aptrishu